packages/realm-server/tests/realm-endpoints/lint-test.ts

```diff
@@ -42,130 +54,821 @@ module(`realm-endpoints/${basename(__filename)}`, function () {
       onRealmSetup,
     });
 
-    test('401 with invalid JWT', async function (assert) {
-      let response = await request
-        .post('/_lint')
-        .set('Authorization', `Bearer invalid-token`)
-        .set('X-HTTP-Method-Override', 'QUERY')
-        .send(`console.log('hi')`);
+    hooks.beforeEach(async function () {
+      testConfig = createTestSuiteConfig({
+        performanceThreshold: 200, // 200ms for CI environment
+        concurrentRequests: 3,
+        includeStressTests: false,
+        includeBackwardCompatibility: true,
+      });
+    });
+
+    // Test Infrastructure Tests
+    module('Test Infrastructure Validation', function () {
+      test('prettier environment is properly configured', async function (assert) {
+        const validation = await validatePrettierEnvironment();
```

Comment:
I’m confused about the structure here, why extract this to a function that’s used one time, that uses a bespoke `warnings`/`errors` structure, instead of asserting directly here?